meson-mode.el

```diff
@@ -892,6 +892,28 @@ Optional SYN-PPSS is the value returned by `syntax-ppss'."
 	    (fspec (alist-get fname meson-builtin-functions)))
       (plist-get fspec :doc)))
 
+(defun meson--find-reference-manual ()
+  "Return the absolute filename of the Meson reference manual or nil."
+  (let ((default-directory meson-markdown-docs-dir))
+    (when-let (manual (seq-find #'file-exists-p '("Reference-manual.md"
+                                                  "Reference-manual.md.gz")))
+      (expand-file-name manual))))
+
+(defun meson--make-lookup-regexp (identifier)
+  "Make regexp for looking up IDENTIFIER in the Meson reference manual."
+  (rx bol (or (+ ?#) ?-) ?  (? ?`)
+      ;; This would be (literal identifier) in Emacs 27.1+.
+      (regexp (regexp-quote identifier))
```

Comment:
Unfortunately, even this is not supported in Emacs 26. There, `regexp` requires string argument and not arbitrary lisp expression as in v27. I tried using `eval` here, but it evaluates the expression at macro expansion time and complains about `identifier` being void. Currently, I don't see a better way than something like `(concat (rx ...) (regexp-quote indetifier) (rx ...))` or using `rx-to-string`, which at runtime. `rx-to-string` would be OK, because it is not performance critical code.

Now, I'm testing this with:
```elisp
(rx-to-string `(seq bol (or (+ ?#) ?-) ?  (? ?`)
      ;; This would be (literal identifier) in Emacs 27.1+.
      (regexp ,(regexp-quote identifier))
      (or ?\( ?` eol)))
```

Reply:
It seems, that `rx` already calls `regexp-quote` on any string literals. I’ve changed my code to this:
```elisp
(rx-to-string
  `(seq bol (or (+ ?#) ?-) ?  (? ?`) ,identifier (or ?\( ?` eol)))
```